payment/controllers.go

```diff
@@ -756,6 +832,47 @@ func VerifyCredential(service *Service) handlers.AppHandler {
 			}
 
 			return handlers.RenderContent(r.Context(), "Credentials successfully verified", w, http.StatusOK)
+		} else if req.Type == "time-limited" {
+			// Presentation includes a token and token metadata test test
+			type Presentation struct {
+				IssuedAt  string `json:"issuedAt"`
+				ExpiresAt string `json:"expiresAt"`
+				Token     string `json:"token"`
+			}
+
+			var bytes []byte
+			bytes, err = base64.StdEncoding.DecodeString(req.Presentation)
+			if err != nil {
+				return handlers.WrapError(err, "Error in decoding presentation", http.StatusBadRequest)
+			}
+
+			var presentation Presentation
+			err = json.Unmarshal(bytes, &presentation)
+			if err != nil {
+				return handlers.WrapError(err, "Error in presentation formatting", http.StatusBadRequest)
+			}
+			timeLimitedSecret := cryptography.NewTimeLimitedSecret([]byte(os.Getenv("BRAVE_MERCHANT_KEY")))
+
+			issuedAt, err := time.Parse("2006-01-02", presentation.IssuedAt)
+			if err != nil {
+				return handlers.WrapError(err, "Error parsing issuedAt", http.StatusBadRequest)
+			}
+			expiresAt, err := time.Parse("2006-01-02", presentation.ExpiresAt)
+			if err != nil {
+				return handlers.WrapError(err, "Error parsing expiresAt", http.StatusBadRequest)
+			}
+
+			verified, err := timeLimitedSecret.Verify(issuedAt, expiresAt, presentation.Token)
```

Comment:
this currently doesn't appear to check against the current time and we're not doing it here so is expiration actually enforced?

Reply:
we should have a unit test that covers the expired case

Reply:
checking expiry: https://github.com/brave-intl/bat-go/pull/863/commits/4bd96fd8b29b50d3302ce21ddf2ee1adb3bbba05 

tests coming next.

Reply:
Test checking expiry here: https://github.com/brave-intl/bat-go/pull/863/commits/512b6945c26ab0643001acfffa7ca9b28f891a4f